packages/x-components/tests/e2e/cucumber/recommendations/recommendations.spec.ts

```diff
@@ -1,22 +1,44 @@
 import { And, Given, Then } from 'cypress-cucumber-preprocessor/steps';
-import { InstallXOptions } from '../../../src/x-installer/x-installer/types';
+import { InstallXOptions } from '../../../../src/x-installer/x-installer/types';
+
+let config: InstallXOptions['xModules'];
 
 Given('following config: max items to store is {int}', (maxItemsToRequest: number) => {
-  const config: InstallXOptions['xModules'] = {
+  config = {
     recommendations: {
       config: {
         maxItemsToRequest
       }
     }
   };
-  cy.visit('/test/recommendations', {
+});
+
+And('suggestions response being mock {string}', (mockName: string) => {
+  cy.intercept('https://api.empathy.co/topRecommendations', req => {
+    import(`./stubs/recommendations.stub`)
+      .then(mock => {
+        return mock[mockName as keyof typeof mock];
+      })
+      .then(mockData => {
+        req.reply(mockData);
+      });
+  }).as('recommendationsApi');
+
+  cy.visit('/test/recommendations?useMockedAdapter=true', {
     qs: {
       xModules: JSON.stringify(config)
     }
   });
 });
 
 // Scenario 1
+Then('number of rows requested is {int}', (maxItemsToRequest: number) => {
+  cy.wait('@recommendationsApi').then(interception => {
+    const interceptedRequestBody = JSON.parse(interception.request.body);
+    expect(interceptedRequestBody.rows).to.equal(maxItemsToRequest);
+  });
```

Comment:
You could refactor this to:

```suggestion
  cy.wait('@interceptedRecommendations').then(({ request }) => {
    const { rows } = JSON.parse(request.body) as TopRecommendationsRequest;
    expect(rows).to.equal(maxItemsToRequest);
  });
```

I think it is more readable using the real request type.